TribalWars/src/Infrastructure/Repository/BuildingRequiredRepository.cs

```diff
@@ -0,0 +1,80 @@
+﻿using ApplicationCore.Entities;
+using ApplicationCore.Enums;
+using ApplicationCore.Interfaces;
+using Infrastructure.Data;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Infrastructure.Repository
+{
+    public class BuildingRequiredRepository : BaseRepository, IBuildingRequiredRepository
+    {
+        public BuildingRequiredRepository(ApplicationDbContext context) : base(context)
+        {
+        }
+
+        public bool HasMaterial(int buildingId, int level, int villageId)
+        {
+            var requiredMaterial = RequiredMaterials(level, buildingId);
```

Comment:
`RequiredMaterials(level, buildingId)` returns `IEnumerable<BuildingRequiredMaterial>`, so the variable should be named `requiredMaterials`